components/Op/OpFormLocation.js

```diff
@@ -1,44 +1,184 @@
+      defaultMessage='Address'
+      description='activity Address label in OpAskForm Form'
     />
     &nbsp;
-    <Tooltip title='set the town or suburb for this activity'>
+    <Tooltip title='set the specific address for this activity'>
       <Icon type='question-circle-o' />
     </Tooltip>
   </span>
 )
-export const OpFormLocation = ({ getFieldDecorator, type, existingLocations, orgMembership }) =>
-  <FormGrid>
-    <DescriptionContainer>
-      <TitleContainer>
-        <h3>
-          <OpTypeLocationPrompt type={type} />
-        </h3>
-      </TitleContainer>
-    </DescriptionContainer>
-    <InputContainer>
-      <MediumInputContainer>
-        <Form.Item label={opLocation}>
-          {getFieldDecorator('locations')(
-            <TagSelect values={existingLocations} placeholder='Select location' />
-          )}
-        </Form.Item>
-        <OpFormOrg getFieldDecorator={getFieldDecorator} type={type} orgMembership={orgMembership} />
-      </MediumInputContainer>
-    </InputContainer>
-  </FormGrid>
+const opSuburb = (
+  <span>
+    <FormattedMessage
+      id='OpShortForm.Suburb'
+      defaultMessage='Suburb'
+      description='activity Suburb label in OpAskForm Form'
+    />
+  </span>
+)
+const opCity = (
+  <span>
+    <FormattedMessage
+      id='OpShortForm.City'
+      defaultMessage='City'
+      description='activity City label in OpAskForm Form'
+    />
+  </span>
+)
+const opPostcode = (
+  <span>
+    <FormattedMessage
+      id='OpShortForm.Postcode'
+      defaultMessage='Postcode'
+      description='activity Postcode label in OpAskForm Form'
+    />
+  </span>
+)
+const opRegion = (
+  <span>
+    <FormattedMessage
+      id='OpShortForm.Region'
+      defaultMessage='Region'
+      description='activity Region label in OpAskForm Form'
+    />
+  </span>
+)
+export const OpFormLocation = ({ getFieldDecorator, type, existingLocations, orgMembership }) => {
+  const [scriptLoaded, setScriptLoaded] = useState(false)
+  const regionRef = useRef(null)
+  const addressRef = useRef(null)
+  const suburbRef = useRef(null)
+  const cityRef = useRef(null)
+  const postcodeRef = useRef(null)
+  const mounted = useRef(false)
+
+  useEffect(() => {
+    mounted.current = true
+    if (window.AddressFinder) {
+      setScriptLoaded(true)
+      return
+    }
+    const script = document.createElement('script')
+    script.src = 'https://api.addressfinder.io/assets/v3/widget.js'
+    script.async = true
+    script.addEventListener('load', () => {
+      if (mounted.current) {
+        setScriptLoaded(true)
+      }
+    })
+    window.document.body.appendChild(script)
+    return () => {
+      mounted.current = false
+      window.document.body.removeChild(script)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (scriptLoaded && mounted.current) {
+      const widget = new window.AddressFinder.Widget(
+        addressRef.current.input,
+        'RGE83VMK4UYHXNJBP69W', // ADDRESSFINDER_KEY
```

Comment:
Is this a private key for our organisation?  If so it should not be put literally into the code base or sent to to the client browser. It would have to be in the env file for the production server and you will have to send the api request first to our server and then onto address finder.

Reply:
Yes, what's the env file we are using? I notice several files like beta.env.enc, alpha.env.encn or development.env.enc....